control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -229,6 +234,16 @@ func Setup(ctx context.Context, opts *operator.HostedClusterConfigOperatorConfig
 	if err := c.Watch(source.Kind[client.Object](opts.CPCluster.GetCache(), &hyperv1.HostedControlPlane{}, eventHandler())); err != nil {
 		return fmt.Errorf("failed to watch HostedControlPlane: %w", err)
 	}
+	p := predicate.NewPredicateFuncs(func(o client.Object) bool {
```

Comment:
Can we add a // coment on why this predicate and watch is needed?

Reply:
Done. Thanks!